mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOps.td

```diff
@@ -668,6 +668,28 @@ def SYCLCallOp : SYCL_Op<"call", [CallOpInterface]> {
   }];
 }
 
+////////////////////////////////////////////////////////////////////////////////
+// accessor.size OPERATION
+////////////////////////////////////////////////////////////////////////////////
+
+def SYCLAccessorSizeOp
+    : SYCLMethodOpInterfaceImpl<"accessor.size", "AccessorType",
+                                ["size"]> {
+  let summary = "Represent the accessor::size";
+  let description = [{
+    This operation returns the number of elements accessible by this SYCL
+    accessor.
+  }];
+
+  let arguments = (ins Arg<AccessorMemRef, "The accessor", [MemRead]>:$Acc,
+                       TypeArrayAttr:$ArgumentTypes,
+                       FlatSymbolRefAttr:$FunctionName,
+                       OptionalAttr<FlatSymbolRefAttr>:$MangledFunctionName,
+                       FlatSymbolRefAttr:$TypeName);
+
+  let results = (outs I64:$Res);
```

Comment:
Minor: I'm wondering if is it possible to set this to `I64` or `I32` easily. The type is `size_t` so that's actually depends on the host definition of it, 64 for what we care right now. If it is trivial, would be good to have this set now rather than chasing bugs later.

Reply:
done.